imagedephi/redact/tiff.py

```diff
@@ -197,10 +198,25 @@ def determine_redaction_operation(
             )
             passes_check = self.passes_type_check(value, rule.valid_data_types, expected_count)
             return "keep" if passes_check else "delete"
-        if rule.action in ["keep", "replace", "delete"]:
+        if rule.action in ["keep", "replace", "delete", "modify_date"]:
             return rule.action
         return "delete"
 
+    def _get_modified_date(self, tiff_date: str) -> str | None:
+        """
+        Given a tiff datestring, return a version set to midnight, January 1st of that year.
+
+        Input should be  a string representing a date (formatted according to the tiff standard,
+        i.e. YYYY:MM:DD HH:MM:DD). If the given string does not conform to the given format,
+        return None.
+        """
+        expected_format = r"\d{4}:\d{2}:\d{2} \d{2}:\d{2}:\d{2}"
```

Comment:
Do we want to add a `$` at the end of this match?  Otherwise, this allows arbitrary extra characters which are dropped (which might be fine -- I could be convinced either way, but we should say we are doing that if we intend it.)

Reply:
See e4814ce. After thinking about it, I came to the conclusion that we should match the start of the string to the end, enforcing the expected pattern as strictly as we can.